This patch fixes compilation of ltrace when you have a native libelf installed.
It comes from openembedded's latest ltrace package.

diff -Nur openembedded.orig/packages/ltrace/files/no-usr-include.patch openembedded/packages/ltrace/files/no-usr-include.patch
--- openembedded.orig/packages/ltrace/files/no-usr-include.patch	1969-12-31 21:00:00.000000000 -0300
+++ openembedded/packages/ltrace/files/no-usr-include.patch	2007-07-02 07:51:11.000000000 -0300
@@ -0,0 +1,16 @@
+--- ltrace-0.4.orig/configure.ac
++++ ltrace-0.4/configure.ac
+@@ -41,13 +41,6 @@
+ dnl   AC_CHECK_FUNCS(gettimeofday strdup strerror strspn)
+ AC_CHECK_FUNCS(getopt getopt_long)
+ 
+-for path in /usr/include/elfutils /usr/local/include/elfutils \
+-	    /usr/include/libelf /usr/local/include/libelf; do
+-	if test -f ${path}/gelf.h; then
+-		CPPFLAGS="$CPPFLAGS -I ${path}"
+-	fi
+-done
+-
+ AC_CHECK_HEADER(gelf.h,,AC_MSG_ERROR([***** gelf.h not found *****]))
+ 		 
+ AC_CHECK_LIB(elf,gelf_getdyn)
diff -Nur openembedded.orig/packages/ltrace/ltrace_0.4.bb openembedded/packages/ltrace/ltrace_0.4.bb
--- openembedded.orig/packages/ltrace/ltrace_0.4.bb	2007-07-02 07:56:28.000000000 -0300
+++ openembedded/packages/ltrace/ltrace_0.4.bb	2007-07-02 07:55:25.000000000 -0300
@@ -6,7 +6,8 @@
 PR = "r0"
 
 SRC_URI = "ftp://ftp.debian.org/debian/pool/main/l/ltrace/ltrace_0.4.orig.tar.gz \
-           ftp://ftp.debian.org/debian/pool/main/l/ltrace/ltrace_0.4-1.diff.gz;patch=1"
+           ftp://ftp.debian.org/debian/pool/main/l/ltrace/ltrace_0.4-1.diff.gz;patch=1 \
+           file://no-usr-include.patch;patch=1"
 inherit autotools
 
 PARALLEL_MAKE = ""
